components/calendar/src/astronomy.rs

```diff
@@ -661,6 +666,36 @@ impl Astronomical {
         div_rem_euclid_f64(n, 360.0).1
     }
 
+    pub fn phasis_on_or_after(date: RataDie, location: Location) -> RataDie {
+        let moon = Self::lunar_phase_at_or_before(0.0, date.as_moment());
+        let age = date - moon.as_rata_die();
+        let tau = if age <= 4 || Self::visible_crescent((date - 1).as_moment(), location) {
+            moon + 29.0 // Next new moon
+        } else {
+            date.as_moment()
+        };
+        next_moment(tau, location, Self::visible_crescent)
+    }
+
+    pub fn phasis_on_or_before(date: RataDie, location: Location) -> RataDie {
+        let moon = Self::lunar_phase_at_or_before(0.0, date.as_moment());
+        let age = date - moon.as_rata_die();
+        let tau = if age <= 3 && !Self::visible_crescent((date).as_moment(), location) {
+            moon - 30.0 // Next new moon
+        } else {
+            moon
+        };
+        next_moment(tau, location, Self::visible_crescent)
+    }
+
+    #[allow(clippy::unwrap_used)]
+    pub fn month_length(date: RataDie, location: Location) -> u8 {
+        let moon = Self::phasis_on_or_after(date + 1, location);
+        let prev = Self::phasis_on_or_before(date, location);
+
+        (moon - prev) as u8
```

Comment:
Nit: Add some debug assertions so we know that the as-cast is safe
```suggestion
        debug_assert!(moon > prev);
        debug_assert!(moon - prev < u8::MAX);
        (moon - prev) as u8
```